File: CF-contest/21-22 ICPC/D_Deletive_Editing.cpp
/*----------------------------------------
    author:  Imam
    created: 07.05.2025 07:26:00
----------------------------------------*/
#include<bits/stdc++.h>
using namespace std;
inline void solve()
{
    string a,b;
    cin>>a>>b;
    unordered_map<char,int>mp1,mp2;
    for(char c:a)mp1[c]++;
    for(char c:b)mp2[c]++;
    int j=0;
    for(int i=0;i<int(a.size());i++){
        if(a[i]==b[j]){
            // cout<<a[i]<<" "<<b[j]<<endl;
            int x=mp1[a[i]];
            int y=mp2[b[j]];
            if(y>x){cout<<"NO\n";return;}
            if(x>y) mp1[a[i]]--;
            else{
                mp1[a[i]]--;
                mp2[b[j]]--;
                j++;
                // cout<<"yes"<<endl;
            }
        }
        else mp1[a[i]]--;
    }
    cout<<(j==int(b.size())?"YES":"NO")<<'\n';
}
int main()
{
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(nullptr); cout.tie(NULL);
    
    int t; cin>>t;
    while(t--) solve();
    return 0;
}

File: CF-contest/21-22 ICPC/D_Deletive_Editing.py
t = int(input())
for _ in range(t):
    a,b = input().split()
    mp1={}
    mp2={}
    for c in a:mp1[c]=mp1.get(c,0)+1
    for c in b:mp2[c]=mp2.get(c,0)+1
    j=0
    for i in range(len(a)):
        if(j<len(b) and a[i]==b[j]):
            x=mp1.get(a[i],0)
            y=mp2.get(b[j],0)
            if(y>x):break
            if(x>y):mp1[a[i]]-=1
            else:
                mp1[a[i]]-=1
                mp2[b[j]]-=1
                j+=1
        else: mp1[a[i]]=mp1.get(a[i],0)-1
    if(j==len(b)):print("YES")
    else: print("NO")